packages/office-ui-fabric-react/src/utilities/positioning/positioning.functions.ts

```diff
@@ -0,0 +1,781 @@
+   * the center of the page.
+   *
+   * @param {IPositionDirectionalHintData} positionData
+   * @param {Rectangle} target
+   * @param {Rectangle} boundingRect
+   * @param {boolean} [coverTarget]
+   * @returns {IPositionDirectionalHintData}
+   */
+  function _getAlignmentData(positionData: IPositionDirectionalHintData, target: Rectangle, boundingRect: Rectangle, coverTarget?: boolean): IPositionDirectionalHintData {
+    if (positionData.isAuto) {
+      positionData.alignmentEdge = getClosestEdge(positionData.targetEdge, target, boundingRect);
+    }
+
+    return positionData;
+  }
+
+  function getClosestEdge(targetEdge: RectangleEdge, target: Rectangle, boundingRect: Rectangle): RectangleEdge {
+    const targetCenter: number = _getCenterValue(target, targetEdge);
+    const boundingCenter: number = _getCenterValue(boundingRect, targetEdge);
+    const { positiveEdge, negativeEdge } = _getFlankingEdges(targetEdge);
+    if (targetCenter <= boundingCenter) {
+      return positiveEdge;
+    } else {
+      return negativeEdge;
+    }
+  }
+
+  export function _positionElementWithinBounds(
+    elementToPosition: Rectangle,
+    target: Rectangle,
+    bounding: Rectangle,
+    positionData: IPositionDirectionalHintData,
+    gap: number,
+    directionalHintFixed?: boolean,
+    coverTarget?: boolean): IElementPosition {
+    let estimatedElementPosition: Rectangle = _estimatePosition(elementToPosition, target, positionData, gap, coverTarget);
+    if (_isRectangleWithinBounds(estimatedElementPosition, bounding)) {
+      return {
+        elementRectangle: estimatedElementPosition,
+        targetEdge: positionData.targetEdge,
+        alignmentEdge: positionData.alignmentEdge
+      };
+    } else {
+      return _adjustFitWithinBounds(elementToPosition, target, bounding, positionData, gap, directionalHintFixed, coverTarget);
+    }
+  }
+
+  export function _finalizeBeakPosition(elementPosition: IElementPosition,
+    positionedBeak: Rectangle): ICalloutBeakPositionedInfo {
+    const targetEdge = elementPosition.targetEdge * -1;
+    // The "host" element that we will use to help position the beak.
+    const actualElement = new Rectangle(0, elementPosition.elementRectangle.width, 0, elementPosition.elementRectangle.height);
+    const returnEdge = elementPosition.alignmentEdge ? elementPosition.alignmentEdge : _getFlankingEdges(targetEdge).positiveEdge;
+    let returnValue: IPartialIRectangle = {};
+
+    returnValue[RectangleEdge[targetEdge]] = _getEdgeValue(positionedBeak, targetEdge);
+    returnValue[RectangleEdge[returnEdge]] = _getRelativeEdgeDifference(positionedBeak, actualElement, returnEdge);
+
+    return {
+      elementPosition: { ...returnValue },
+      closestEdge: getClosestEdge(elementPosition.targetEdge, positionedBeak, actualElement),
+      targetEdge: targetEdge
+    };
+  }
+
+  export function _positionBeak(beakWidth: number,
```

Comment:
exported functions are public, not underscored.

I would also suggest to move publics to the top if possible. That way you read the publics, which are the important parts.

Or just make them all public, so that they can be unit tested, and choose which ones you want to export publicly from an index file.